libraries/render-utils/src/text/Font.cpp

```diff
@@ -347,38 +359,8 @@ void Font::buildVertices(Font::DrawInfo& drawInfo, const QString& str, const glm
         if (!isNewLine) {
             for (auto c : token) {
                 auto glyph = _glyphs[c];
-                quint16 verticesOffset = numVertices;
-
-                QuadBuilder qd(glyph, advance - glm::vec2(0.0f, _ascent), scale, enlargeForShadows);
-                drawInfo.verticesBuffer->append(qd);
-                numVertices += VERTICES_PER_QUAD;
-
-                // Sam's recommended triangle slices
-                // Triangle tri1 = { v0, v1, v3 };
-                // Triangle tri2 = { v1, v2, v3 };
-                // NOTE: Random guy on the internet's recommended triangle slices
-                // Triangle tri1 = { v0, v1, v2 };
-                // Triangle tri2 = { v2, v3, v0 };
-
-                // The problem here being that the 4 vertices are { ll, lr, ul, ur }, a Z pattern
-                // Additionally, you want to ensure that the shared side vertices are used sequentially
-                // to improve cache locality
-                //
-                //  2 -- 3
-                //  |    |
-                //  |    |
-                //  0 -- 1
-                //
-                //  { 0, 1, 2 } -> { 2, 1, 3 }
-                quint16 indices[NUMBER_OF_INDICES_PER_QUAD];
-                indices[0] = verticesOffset + 0;
-                indices[1] = verticesOffset + 1;
-                indices[2] = verticesOffset + 2;
-                indices[3] = verticesOffset + 2;
-                indices[4] = verticesOffset + 1;
-                indices[5] = verticesOffset + 3;
-                drawInfo.indicesBuffer->append(sizeof(indices), (const gpu::Byte*)indices);
-                drawInfo.indexCount += NUMBER_OF_INDICES_PER_QUAD;
+
+                glyphsAndCorners.emplace_back(glyph, advance - glm::vec2(0.0f, _ascent));
```

Comment:
we don’t know how long a line will be until we finish looping over all the characters.  so, we first build all the glyph/position pairs in the same way as before